lib/vector-core/src/event/value.rs

```diff
@@ -49,7 +49,9 @@ impl From<Bytes> for Value {
 
 impl<T: Into<Value>> From<Vec<T>> for Value {
     fn from(set: Vec<T>) -> Self {
-        Value::from_iter(set.into_iter().map(|v| v.into()))
+        set.into_iter()
+            .map(::std::convert::Into::into)
+            .collect::<Value>()
```

Comment:
Can this be `.collect::<Self>()`?

Reply:
I bet it could. Let me give that a shot.